examples/face_recognition_11/face_recognition_11/evaluator.py

```diff
@@ -0,0 +1,402 @@
+            normalization_factor,
+        )
+        Δ_left_eye, norm_Δ_left_eye = compute_distances(
+            ground_truth.keypoints.points[1],
+            inference.keypoints.points[1],
+            normalization_factor,
+        )
+        Δ_right_eye, norm_Δ_right_eye = compute_distances(
+            ground_truth.keypoints.points[2],
+            inference.keypoints.points[2],
+            normalization_factor,
+        )
+        Δ_left_mouth, norm_Δ_left_mouth = compute_distances(
+            ground_truth.keypoints.points[3],
+            inference.keypoints.points[3],
+            normalization_factor,
+        )
+        Δ_right_mouth, norm_Δ_right_mouth = compute_distances(
+            ground_truth.keypoints.points[4],
+            inference.keypoints.points[4],
+            normalization_factor,
+        )
+        distances = np.array([Δ_left_eye, Δ_right_eye, Δ_nose, Δ_left_mouth, Δ_right_mouth])
+        mse, nmse = calculate_mse_nmse(distances, normalization_factor)
+
+    if not bbox_fte and (inference.keypoints is None or nmse >= configuration.nmse_threshold):
+        keypoint_fta = True
+
+    # Stage 3: Recognition
+    pair_samples = list()
+    count_fnm, count_fm, count_tm, count_tnm = 0, 0, 0, 0
+    for i, (is_same, similarity) in enumerate(zip(ground_truth.matches, inference.similarities)):
+        is_match, is_false_match, is_false_non_match, failure_to_enroll = False, False, False, False
+        if similarity is None:
+            if is_same:
+                is_false_non_match = True
+            failure_to_enroll = True
+        elif similarity > threshold:  # match
+            if is_same:
+                is_match = True
+            else:
+                is_false_match = True
+        else:  # no match
+            if is_same:
+                is_false_non_match = True
+
+        pair_sample = PairSample(
+            locator=test_sample.pairs[i].locator if test_sample else "",
+            is_match=is_match,
+            is_false_match=is_false_match,
+            is_false_non_match=is_false_non_match,
+            failure_to_enroll=failure_to_enroll,
+            similarity=similarity,
+        )
+        pair_samples.append(pair_sample)
+        count_fnm += is_false_non_match
+        count_fm += is_false_match
+        count_tm += is_match
+        count_tnm += not is_false_non_match and not is_false_match and not is_match
+
+    return TestSampleMetrics(
+        pair_samples=pair_samples,
+        count_FNM=count_fnm,
+        count_FM=count_fm,
+        count_TM=count_tm,
+        count_TNM=count_tnm,
+        similarity_threshold=threshold,
+        bbox_IoU=iou_value if iou_value is not None else 0.0,
+        bbox_TP=[inference.bbox] if tp and inference.bbox is not None else [],
+        bbox_FP=[inference.bbox] if fp and inference.bbox is not None else [],
+        bbox_FN=[inference.bbox] if (not tp and not fp) and inference.bbox is not None else [],
```

Comment:
How can an inference be a false negative? Usually, in object detection false negatives are ground truth bounding boxes that are not matched with a corresponding prediction.

Reply:
It almost feels like we want `bbox_success` and `bbox_failure` here instead of TP/FP/FN. I'll defer to @y27choi here though

Reply:
Thanks for flagging.  I think FN should be GT bbox that are not matched, but `bbox_success` and `bbox_failure` make sense here. Will these names work with the default coloring logic?

Reply:
Yep, default coloration logic will work as expected using those names.

I've been on the fence about making this suggestion because TP/FP/FN are so standard for OD, but it really does feel like success/failure is more relevant here when there's always a single box to predict in the image. If we see this extending beyond that, then let's keep TP/FP/FN, and if so let's update the FN logic to be unmatched GTs.

Reply:
> let's keep TP/FP/FN, and if so let's update the FN logic to be unmatched GTs

I'm leaning towards just updating FN logic here. @a75w-kolena  and I talked about potentially adding multiple faces inference in the future and keeping it as TP/FP/FN makes sense in such case.